cunit/charset.testc

```diff
@@ -1213,6 +1213,10 @@ static void test_extract(void)
     // XXX img with alt but no src
     TESTCASE("<img alt=\"Nothing\"/>", "us-ascii", ENCODING_NONE, "HTML",
              "(NOTHING)");
+    // XXX multiple hrefs in same A tag (invalid!)
+    TESTCASE("<a href=\"https://foo/bar\" href=\"https://baz/qum\"/>",
+             "us-ascii", ENCODING_NONE, "HTML",
+             "what do we want here?");
```

Comment:
This is required to make the HTML extractor use the last value (which also solve the same issue for "alt"):

     diff --git a/lib/charset.c b/lib/charset.c
     index d3034b6b3c..8704ef0f77 100644
     --- a/lib/charset.c
     +++ b/lib/charset.c
     @@ -1323,6 +1323,10 @@ static void html_attr_start(struct striphtml_state *s)
                  s->attr.typ = HATTR_ALT;
              }
          }
     +
     +    if (s->attr.typ != HATTR_NONE) {
     +        dynarray_truncate(&s->attr.vals[s->attr.typ], 0);
     +    }
      }

      static void html_attr_putc(struct striphtml_state *s, uint32_t c)

I say we just do this, rather than supporting the edge case of finding invalid html.